src/threshold_decryption.rs

```diff
@@ -115,22 +115,32 @@ impl<N: NodeIdT> ThresholdDecryption<N> {
     /// Sets the ciphertext, sends the decryption share, and tries to decrypt it.
     /// This must be called exactly once, with the same ciphertext in all participating nodes.
     /// If we have enough shares, outputs the plaintext.
-    pub fn set_ciphertext(&mut self, ct: Ciphertext) -> Result<Step<N>> {
+    pub fn set_message(&mut self, ct: Ciphertext) -> Result<()> {
         if self.ciphertext.is_some() {
             return Err(Error::MultipleInputs(Box::new(ct)));
         }
+        self.ciphertext = Some(ct);
+        Ok(())
+    }
+
+    /// Sends our signature shares, and if we have collected enough, returns the decrypted message.
+    /// Returns an empty step if the ciphertext hasn't been received yet.
+    pub fn sign(&mut self) -> Result<Step<N>> {
+        let ct = match self.ciphertext {
+            None => return Ok(Step::default()),
```

Comment:
I think we should return an error here: The user is not supposed to call this method before `set_ciphertext`.